src/dmd/mtype.d

```diff
@@ -4685,6 +4685,14 @@ extern (C++) final class TypeFunction : TypeNext
             match = MATCH.convert; // match ... with a "conversion" match level
         }
 
+        if (parameterList.varargs == VarArg.none && nparams > nargs && !parameterList[nargs].defaultArg)
```

Comment:
Can you add a link to https://issues.dlang.org/show_bug.cgi?id=22997 in a comment? It's not self-explanatory why this branch is here.